plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerBehaviour.java

```diff
@@ -63,7 +66,9 @@
   private LocalAppEngineExitListener localAppEngineExitListener;
   private AppEngineDevServer devServer;
   private Process devProcess;
-  private int port = -1;
+
+  @VisibleForTesting int serverPort = -1;
```

Comment:
We always pull the server port value set by the user first. The logic works differently from the admin port; we never change it to something else. That said, I think -1 is better than 8080. To me, 8080 gives a false impression that we will by default use 8080.